backend/src/test/java/de/neuefische/backend/service/CovidTestServiceTest.java

```diff
@@ -0,0 +1,87 @@
+package de.neuefische.backend.service;
+
+import de.neuefische.backend.dto.CovidTestDto;
+import de.neuefische.backend.model.CovidTest;
+import de.neuefische.backend.model.TestType;
+import de.neuefische.backend.repo.CovidTestRepo;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+
+class CovidTestServiceTest {
+
+    private final CovidTestRepo covidTestRepo = mock(CovidTestRepo.class);
+    private final CovidTestService covidTestService = new CovidTestService(covidTestRepo);
+
+    @Test
+    void getCovidTestsTest() {
+        // GIVEN
+        List<CovidTest> expectedCovidTests = List.of(CovidTest.builder()
+                .testType(TestType.ANTIGEN_NASAL)
+                .dateTime(LocalDateTime.of(2021, 11, 15, 7, 5))
+                .build());
+        when(covidTestRepo.findAll()).thenReturn(expectedCovidTests);
+
+        // WHEN
+        List<CovidTest> actualCovidTests = covidTestService.getCovidTests();
+
+        // THEN
+        assertEquals(expectedCovidTests, actualCovidTests);
+        verify(covidTestRepo).findAll();
+    }
+
+
+
+    @Test
+    @DisplayName("addCovidTest should add a new covidTest to the database")
+    void addCovidTestTest() {
+        // GIVEN
+        CovidTestDto expectedCovidTestDto = CovidTestDto.builder()
+                .testType(TestType.ANTIGEN_NASAL)
+                .dateTime(LocalDateTime.of(2021, 11, 15, 7, 5))
+                .build();
+
+        CovidTest expectedCovidTest = CovidTest.builder()
+                .testType(TestType.ANTIGEN_NASAL)
+                .dateTime(LocalDateTime.of(2021, 11, 15, 7, 5))
+                .build();
+
+        when(covidTestRepo.save(expectedCovidTest)).thenReturn(expectedCovidTest);
+
+        // WHEN
+        CovidTest actualCovidTest = covidTestService.addCovidTest(expectedCovidTestDto);
+
+        // THEN
+        assertEquals(expectedCovidTest, actualCovidTest);
+        verify(covidTestRepo).save(expectedCovidTest);
```

Comment:
Same as above.
But in this case, I would change the `when` statement.
```java
when(covidTestRepo.save(covidTestToSave)).thenReturn(expectedCovidTest);

// WHEN
CovidTest actualCovidTest = covidTestService.addCovidTest(expectedCovidTestDto);

// THEN
assertSame(expectedCovidTest, actualCovidTest);
```
`covidTestToSave` and `expectedCovidTest` need to be different instances in that case. That way you would make sure, that your service really returns the object, that was returned by the save method.